shapeshift/web
# feat: foxy opportunities full multi-account support

## Description

- Overview
  - [x] Account selection
  - [x] Use selected account data
  - [ ] Route from DeFi cards - TBD in https://github.com/shapeshift/web/issues/2615
- [x] Deposit
  - [x] Approve
  - [x] Confirm
  - [x] Use selected account data in all steps
  - [x] Poll (regression test against account 0 on prod) 
- [x] Withdraw
  - [x] Approve
  - [x] Confirm 
  - [x] Use selected account data in all steps
  - [x] Poll (regression test against account 0 on prod)
 

## Notice




## Pull Request Type


## Issue (if applicable)

- relates to the closed https://github.com/shapeshift/web/issues/2611
- relates to https://github.com/shapeshift/web/issues/2615
- relates to https://github.com/shapeshift/web/issues/2730
- closes https://github.com/shapeshift/web/issues/2758

## Risk

Tested with MetaMask (which has only account 0 support) but this could bring regressions forFOXy modal for single accounts, test accordingly

## Testing (with KeepKey and Native)

- Add an Ethereum account and send some ETH and FOX
- Stake in some opportunity (or skip this step if already staked in FOXy), reload the app (websocket is currently borked) and click on the opportunity card for it
- Select account 1 and click deposit
- Notice account 1 is pre-selected at Deposit step, with the right FOX amount to deposit 
- Deposit some more of FOX, sign and broadcast it - it should be deposited from account 1
- Disconnect and reconnect wallet, then reopen the overview modal and select withdraw with account 1 selected in overview
- Notice account 1 is pre-selected at Withdraw step, with the right amount of FOXy available to withdraw
- Withdraw some of your FOXy - it should be withdrawn from account 1
- Disconnect and reconnect wallet, then reopen the FOXy overview modal with account 1 selected

### Engineering



- Refer to top-level testing steps
- Double-check the logic looks sound in deposit withdraw and claim for all steps

### Operations

- Refer to top-level testing steps



## Screenshots

<img width="535" alt="image" src="https://user-images.githubusercontent.com/17035424/191129554-b9b13fc3-7cb9-40c3-84fa-0c3ec74df03b.png">
<img width="534" alt="image" src="https://user-images.githubusercontent.com/17035424/191129492-f88f2caa-988c-43c6-ace9-a853823de568.png">
<img width="531" alt="image" src="https://user-images.githubusercontent.com/17035424/191129517-6c8c2513-fc4c-4165-87a2-0ba058db6428.png">
